packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts

```diff
@@ -17,60 +17,58 @@ import {
 import { isHostElement } from '../../faux-shadow/shadow-root';
 import { getEventListenerWrapper } from '../../shared/event-target';
 
-function isValidEventListener(listener: EventListenerOrEventListenerObject): boolean {
-    return (
-        isFunction(listener) ||
-        (!isNull(listener) && isObject(listener) && isFunction(listener.handleEvent))
-    );
-}
-
-function addEventListener(
+function patchedAddEventListener(
     this: EventTarget,
-    type: string,
-    listener: EventListenerOrEventListenerObject,
-    optionsOrCapture?: boolean | AddEventListenerOptions
+    _type: string,
+    _listener: EventListenerOrEventListenerObject,
+    _optionsOrCapture?: boolean | AddEventListenerOptions
 ) {
-    // TODO [#2134]: Delegate this validation to the browser instead of silently swallowing
-    if (!isValidEventListener(listener)) {
-        return;
-    }
+    const args = [...arguments];
     if (isHostElement(this)) {
-        addCustomElementEventListener(this, type, listener);
-    } else {
-        const wrapperFn = getEventListenerWrapper(listener);
-        nativeAddEventListener.call(this, type, wrapperFn, optionsOrCapture);
+        // Typescript does not like it when you treat the `arguments` object as an array
+        // @ts-ignore type-mismatch
+        return addCustomElementEventListener(this, ...args);
```

Comment:
Let's just go with `return addCustomElementEventListener(this, ...arguments);` because the following combined operations is [very expensive](https://babeljs.io/repl/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABMOcAUBKRBvAUIxCBAZykQEMAnAc2MQF5EBtAOjaupAFsBTMKYgF0A3PkQAjKmjYsOxDKIC-uUJFgIJUrNkVA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ces2015%2Creact&prettier=false&targets=&version=7.12.12&externalPlugins=) in compat mode.
```
const args = [...arguments];
fn(...args);
```
As part of this change, we can convert `arguments` into an Array after handling the host element case, since that operation itself has a performance cost.

We should also use the cached `ArraySlice` from `@lwc/shared` instead of the spread operator to minimize the possibility of invoking a patched version of `Array.prototype.slice`.